### PR TITLE
lib-injection: fail fast in tests

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description

We were missing the `-e` flag in functions.sh which means that failing commands were silently passing.

By adding this flag it was discovered that k8s commands were silently failing as the node was not yet ready. A wait is added for the node to be ready which solves a lot of the race condition flakiness we were seeing before.

Also remove some of the python variants as they are tested and run in the dd-trace-py CI.

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: lib-injection/src/test/shell/functions.sh [hunk 1]
```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+# Exit early for any failed commands
+set -e
+# Print commands that are run
+set -x
```
Comment (robertomonteromiguel) [inline lib-injection/src/test/shell/functions.sh:6]: same set -ex

**Review by robertomonteromiguel — APPROVED** [review]
In my opinion the changes improve the cleanliness of the code, but I am concerned about the timing related changes (initialDelaySeconds, periodSeconds). My recommendation is to run the tests several times with all variants, because some errors may appear occasionally.

**Comment by Kyle-Verhoog** [comment]
Thanks @robertomonteromiguel, yeah I think with the `kubectl wait --for=condition=Ready nodes --all --timeout=5m` check then we don't need the conservative readiness probes. I will rerun a few times to confirm though.

**Comment by Kyle-Verhoog** [comment]
Looks like we're pretty consistently passing after 5 runs

**Pull request merged**